core/src/program.rs

```diff
@@ -201,10 +212,48 @@ impl<EC: EvalCache> Program<EC> {
         self.vm.eval_full(t, &initial_env).map_err(|e| e.into())
     }
 
-    /// Same as `eval`, but proceeds to a full evaluation.
-    /// Skips record fields marked `not_exported`
-    pub fn eval_full_for_export(&mut self) -> Result<RichTerm, Error> {
-        let (t, initial_env) = self.prepare_eval()?;
+    /// Same as `eval`, but proceeds to a full evaluation. Optionally take a set of overrides that
+    /// are to be applied to the term (in practice, to be merged with).
+    ///
+    /// Skips record fields marked `not_exported`.
+    ///
+    /// # Arguments
+    ///
+    /// - `override` is a list of overrides, where the first element is a field path represented as an
+    ///   array of strings, and the second is Nickel source code as string. Each override is imported
+    ///   in a separate in-memory source, for complete isolation (this way, overrides can't
+    ///   accidentally or intentionally capture other fields of the configuration). A stub record is
+    ///   then built, which has all fields defined by `overrides`, and values are an import referring
+    ///   to the corresponding isolated value. This stub is finally merged with the current program
+    ///   before being evaluated for import.
+    pub fn eval_full_for_export(
+        &mut self,
+        overrides: Option<impl IntoIterator<Item = FieldOverride>>,
+    ) -> Result<RichTerm, Error> {
+        let (t, initial_env) = match overrides {
+            None => self.prepare_eval()?,
+            Some(overrides) => {
+                let mut record = builder::Record::new();
+
+                for ovd in overrides {
+                    let value_file_id = self
+                        .vm
+                        .import_resolver_mut()
+                        .add_string(format!("<override {}>", ovd.path.join(".")), ovd.value);
+                    self.vm.prepare_eval(value_file_id)?;
+                    record = record
+                        .path(ovd.path)
+                        .priority(ovd.priority)
+                        .value(Term::ResolvedImport(value_file_id));
+                }
+
+                let (t, initial_env) = self.prepare_eval()?;
+                let built_record = record.build();
+                let wrapper =
+                    mk_term::op2(BinaryOp::Merge(Label::default().into()), t, built_record);
```

Comment:
I'm not sure if we can do better than the `Label::default()` here for the merge label. But it certainly isn't ideal if the error reporting path ever ends up looking at it. At the moment we might not be able to hit that, but it's certainly a future footgun.

Reply:
We can indeed hit that error, if we try to override something that is already `force`, which can cause various merge issues. In my tests it's not too bad, but this is probably sheer luck (in that the generated default label always point to the first source loaded, which happens to be the original program), and could probably go very wrong.

That being said, I don't think we can really fix that here. The "all labels must refer to a precise span" policy has probably lived for too long, now that we're generating terms programmatically, and we probably can't avoid sprinkling a bunch of optionals (in practice, probably `RawSpan -> TermPos`) here and there.

Reply:
Documented in 7882a96d